python/foglamp/plugins/south/cc2650async/cc2650async.py

```diff
@@ -336,9 +336,43 @@ def plugin_reconfigure(handle, new_config):
     """
     _LOGGER.info("Old config for CC2650ASYN plugin {} \n new config {}".format(handle, new_config))
 
-    new_handle = plugin_init(new_config)
-    return new_handle
+    # Find diff between old config and new config
+    diff = list()
+    for key in new_config:
+        if key in handle:
+            if handle[key] != new_config[key]:
+                diff.append(key)
+        else:
+            diff.append(key)
+
+    # Plugin should re-initialize and restart if key configuration is changed
+    if 'bluetoothAddress' in diff:
+        # TODO: Investigate if a common stop_plugin() method, shared by plugin_shutdown(), required.
+        if 'tag' in handle:
+            bluetooth_adr = handle['bluetoothAddress']['value']
+            tag = handle['tag']
+
+            # Disable sensors
+            tag.char_write_cmd(handle['characteristics']['temperature']['configuration']['handle'], char_disable)
+            tag.char_write_cmd(handle['characteristics']['luminance']['configuration']['handle'], char_disable)
+            tag.char_write_cmd(handle['characteristics']['humidity']['configuration']['handle'], char_disable)
+            tag.char_write_cmd(handle['characteristics']['pressure']['configuration']['handle'], char_disable)
+            tag.char_write_cmd(handle['characteristics']['movement']['configuration']['handle'], movement_disable)
+
+            # Disable notification
+            for notification_handle in handle['notification_handles']:
+                tag.char_write_cmd(notification_handle, notification_disable)
 
+            tag.disconnect()
+            _LOGGER.info('SensorTagCC2650 (async) {} Disconnected.'.format(bluetooth_adr))
+
+        new_handle = plugin_init(new_config)
+        new_handle['restart'] = 'yes'
+        _LOGGER.info("Restarting CC2650ASYN plugin due to change in configuration keys [{}]".format(', '.join(diff)))
+    else:
+        new_handle = copy.deepcopy(handle)
+        new_handle['restart'] = 'no'
```

Comment:
better to set True / False ? as its not saved to config (which accepts literal) but run time decision

Reply:
For better self documentation purpose, yes/no has been chosen over True/False.